Core/GDCore/Project/Variable.h

```diff
@@ -183,20 +218,53 @@ class GD_CORE_API Variable {
   void RemoveRecursively(const gd::Variable& variableToRemove);
   ///@}
 
-  /** \name Serialization
-   * Methods used when to load or save a variable to XML.
+  /** \name Array
+   * Methods used when the variable is considered as an array.
    */
   ///@{
+
+  /**
+   * \brief Return true if the variable is an array and has the specified
+   * index.
+   */
+  bool HasIndex(const size_t index) const;

```

Comment:
I think we should remove `HasIndex` from the public interface and just manually make the check for type is array and `childrenList.size()` whenever it's needed.